### ISSUE TITLE
Type for each timezone

### ISSUE THREAD

**Issue by fenhl** [chronotope/chrono-tz#30]
This feature was removed in #1. While I understand the reason for needing a single type that can dynamically represent different time zones, a project I'm currently working on needs to statically mark a value as being in a specific timezone:

```rust
#[derive(Deserialize)]
struct Event {
    pub start: DateTime<Europe::Berlin>,
    pub end: DateTime<Europe::Berlin>,
    // and some irrelevant fields ...
}
```

This is because the timestamps I'm deserializing are not marked with any timezone but the documentation specifies that they are to be interpreted as local times in Germany (and that no DST-ambiguous timestamps occur).

It would be great if this library could support both use-cases.

**Comment by quodlibetor** [comment]
[chrono-tz](https://docs.rs/chrono-tz/0.5.1/chrono_tz/) should support this.

**Comment by asnimansari** [comment]
+1

**Comment by lukesneeringer** [comment]
Yes, this would be extremely useful. The following simple use case is, as best as I can tell, impossible:

```rs
use chrono::DateTime;
use chrono_tz::US::Eastern;

struct Foo {
  create_time: DateTime<Eastern>,
}
```

As best as I can tell, it is only feasible to store DateTimes in Utc/FixedOffset/Local in a struct.

**Comment by robertbastian** [comment]
This would also provide dead-code elimination, so only data for time zones that are actually used would be included. Important for binary sizes on embedded.
